src/sync.js

```diff
@@ -184,6 +184,11 @@ class Sync {
       case side == null:
         this.pouch.setLocalSeq(change.seq, callback)
         break
+      case (sideName === 'remote' && doc.toBeTrashed && !inRemoteTrash(doc)):
+        // File or folder was just deleted locally
+        this.trashLaterWithParentOrByItself(doc, side)
+        done()
```

Comment:
I think that just adding a TODO for the moment is OK.

Yes, we have to call `done()` to move to the next changes. But we can modify the document (`doc.toReallyTrash = true`) to make it pop again in sync later. We have to figure the details (when to do that, how to preserve the side that have to apply this, etc.) and I suspect it won't be easy.